core/sdk-core/src/main/java/software/amazon/awssdk/core/http/ExecutionContext.java

```diff
@@ -37,10 +36,10 @@
     private final ExecutionAttributes executionAttributes;
 
     private ExecutionContext(final Builder builder) {
-        this.signer = Validate.paramNotNull(builder.signer, "signer");
```

Comment:
I added them originally, but in hindsight they just made testing more difficult. 

I ended up not needing them removed, because I changed the way I was testing this change, so I could add them back in again.